config/class-site-details-index.php

```diff
@@ -349,32 +354,127 @@ public function get_site_details() {
 	}
 
 	/**
-	 * Builds the site details structure and then puts it into logstash
-	 * and sends it to the site details service
+	 * Builds the site details and sends it to the site details service.
 	 */
 	public function put_site_details() {
 		$site_details = $this->get_site_details();
-		$url          = null;
-		$token        = null;
+		$sync_data    = get_option( self::SYNC_DATA_OPTION, [] );
+		$sync_data    = is_array( $sync_data ) ? $sync_data : [];
+
+		$sync_type = $this->determine_sync_type( $site_details, $sync_data );
+
+		// Run a heartbeat sync.
+		if ( 'heartbeat' === $sync_type ) {
+			$success = $this->send_sync( '/sites/heartbeat', [
+				'client_site_id' => $site_details['client_site_id'],
+				'blog_id'        => $site_details['core']['blog_id'],
+				'timestamp'      => $site_details['timestamp'],
+			] );
+
+			if ( $success ) {
+				$sync_data['last_synced'] = $site_details['timestamp'];
+				update_option( self::SYNC_DATA_OPTION, $sync_data, false );
+			}
+		}
+
+		// Run a full sync.
+		if ( 'full' === $sync_type ) {
+			$success = $this->send_sync( '/sites', $site_details );
 
-		if ( defined( 'VIP_SERVICES_AUTH_TOKENS' ) && ! empty( VIP_SERVICES_AUTH_TOKENS ) ) {
-			$auth_token_details = json_decode( base64_decode( VIP_SERVICES_AUTH_TOKENS ), true );
-			$url                = $auth_token_details['site']['vip-site-details']['url'] ?? null;
-			$token              = $auth_token_details['site']['vip-site-details']['token'] ?? null;
+			if ( $success ) {
+				$timestamp = $site_details['timestamp'];
+
+				// Stagger the future syncs if this is the first time we're setting the option.
+				if ( empty( $sync_data['last_full_synced'] ) ) {
+					$sync_data['last_full_synced'] = wp_rand( $timestamp - self::DAY_IN_MS, $timestamp );
+					$sync_data['last_synced']      = wp_rand( $timestamp - self::MINUTE_IN_MS * 25, $timestamp );
+				} else {
+					$sync_data['last_full_synced'] = $timestamp;
+					$sync_data['last_synced']      = $timestamp;
+				}
+
+				$sync_data['last_sync_hash'] = $this->get_site_details_data_hash( $site_details );
+				update_option( self::SYNC_DATA_OPTION, $sync_data, false );
+			}
 		}
+	}
 
-		if ( $url && $token ) {
-			$args = array(
-				'method'  => 'PUT',
-				'body'    => wp_json_encode( $site_details ),
-				'headers' => array(
-					'Authorization' => 'Bearer ' . $token,
-					'Content-Type'  => 'application/json',
-				),
-			);
+	/**
+	 * Determine if we need a full sync, a heartbeat, or none at all.
+	 */
+	private function determine_sync_type( $site_details, $sync_data ) {
+		$current_timestamp        = $site_details['timestamp'];
+		$last_sync_timestamp      = $sync_data['last_synced'] ?? 0;
```

Comment:
I'm seeing this same statement twice, once declared here and one below the comment `// Send a heartbeat if it's been more than 25 minutes (1/3 of the stale threshold).`